src/OrchardCore.Modules/OrchardCore.Users/Liquid/UserFilters.cs

```diff
@@ -31,21 +29,10 @@ public static async ValueTask<FluidValue> HasClaim(FluidValue input, FilterArgum
                     return BooleanValue.True;
                 }
 
-                if (string.Equals(claimType, Permission.ClaimType, StringComparison.OrdinalIgnoreCase))
+                if (string.Equals(claimType, Permission.ClaimType, StringComparison.OrdinalIgnoreCase) &&
```

Comment:
But this would be removed if we have a custom filter for authorization, right?

Reply:
I don't think we would. Because we should still return true if the user is an admin. `has_authorization` would use the `IAutheorizationService` to do similar check without evaluating the claims explicitly. 

If we need to remove it, we'll have to document it as a breaking change for 3.0 and offer `has_authorization` as a better way to authorize user

Reply:
> Because we should still return true if the user is an admin

In 2.x because it would be a breaking change otherwise. So we can remove it in 3.0.

Reply:
So are you saying to keep this and at somepoint add `has_authorization`. then in 3.0, we can remove this check from has_permission, correct?

Reply:
@sebastienros actually we already have `has_permission` filter that one can already use. So I don't think there is a need to add `has_authorization`

Reply:
I also think this should be removed e.g. in 3.0, but lets keep it for backward compatibility until then.

Reply:
I added a note so we can remove it in 3.0